uaclient/clouds/gcp.py

```diff
@@ -42,11 +42,40 @@ def __init__(self):
     # mypy does not handle @property around inner decorators
     # https://github.com/python/mypy/issues/1362
     @property  # type: ignore
-    @util.retry(HTTPError, retry_sleeps=[1, 2, 5])
+    @util.retry(exceptions.GCPProAccountError, retry_sleeps=[1, 2, 5])
     def identity_doc(self) -> Dict[str, Any]:
-        url_response, _headers = util.readurl(
-            TOKEN_URL, headers={"Metadata-Flavor": "Google"}
-        )
+        try:
+            headers = {"Metadata-Flavor": "Google"}
+            url_response, _headers = util.readurl(TOKEN_URL, headers=headers)
+        except URLError as e:
+            body = None
+            if hasattr(e, "body"):
+                body = getattr(e, "body", None)
+            elif hasattr(e, "read"):
+                reader = getattr(e, "read")()
+                body = reader.decode("utf-8")
+            error_desc = None
+            if body:
+                try:
+                    error_details = json.loads(
+                        body, cls=util.DatetimeAwareJSONDecoder
+                    )
+                except ValueError:
+                    error_details = None
+                if error_details:
+                    error_desc = error_details.get("error_description", None)
+            msg = error_desc if error_desc else getattr(e, "msg", None)
```

Comment:
When would `e` have a `msg` property?

Reply:
Oh interesting. You're right that it uses the `msg` property internally seen here: https://github.com/python/cpython/blob/3.10/Lib/urllib/error.py#L39=

But it isn't documented anywhere here: https://docs.python.org/3/library/urllib.error.html?highlight=httperror#urllib.error.HTTPError

My guess is that they want you to use the `reason` property instead, which just returns `msg` https://github.com/python/cpython/blob/3.10/Lib/urllib/error.py#L61=

So can we use `reason` instead of `msg` here?

Reply:
Oh interesting. You're right that it uses the `msg` property internally seen here: https://github.com/python/cpython/blob/3.10/Lib/urllib/error.py#L39=

But it isn't documented anywhere here: https://docs.python.org/3/library/urllib.error.html?highlight=httperror#urllib.error.HTTPError

My guess is that they want you to use the `reason` property instead, which just returns `msg` https://github.com/python/cpython/blob/3.10/Lib/urllib/error.py#L61=

So can we use `reason` instead of `msg` here?

Reply:
Oh makes sense, just found the thread proposing inclusion of `reason` property [here](https://bugs.python.org/issue13211)